logic.py

```diff
@@ -0,0 +1,72 @@
+import io
+import chess.pgn
+from utils import get_sharpest_lines_from_fen
+
+def get_san(node, move):
+    return node.board().san(move)
+
+# TODO: Be able to infer is_user_white in the future from repertoire / get request.
```

Comment:
Can't think of a straightforward way of figuring out if user is white/black apart from directly asking them.

Reply:
Perhaps we can do it when the user is importing the PGN instead